src/containers/flows/TransferOwnership/TransferOwnership.tsx

```diff
@@ -66,120 +74,48 @@ const TransferOwnership: React.FC<TransferOwnershipProps> = ({
-					</p>
-					{isLoading && <p>Transaction pending user confirmation.</p>}
-					<div className={styles.Buttons}>
-						{!isLoading && (
-							<FutureButton
-								style={{ textTransform: 'uppercase' }}
-								alt
-								glow
-								onClick={cancel}
-							>
-								Cancel
-							</FutureButton>
-						)}
-						<FutureButton
-							style={{ textTransform: 'uppercase' }}
-							glow
-							loading={isLoading}
-							onClick={submitTransfer}
-						>
-							Transfer
-						</FutureButton>
-					</div>
-				</div>
-			</Overlay>
-
-			<Overlay open={!hasAccepted} onClose={() => onTransfer()}>
-				<div
-					className={`${styles.TransferOwnership} blur border-rounded border-primary`}
-				>
-					<div className={styles.Header}>
-						<h1 className={`glow-text-white`}>Transfer Ownership</h1>
-					</div>
-
-					<hr className="glow" />
-
-					<div
-						className={styles.Section}
-						style={{ display: 'flex', padding: '0 37.5px' }}
-					>
-						<div className={styles.NFT}>
-							<NFTMedia alt="Bid NFT preview" ipfsUrl={image} size="small" />
-						</div>
-
-						<div className={styles.Details}>
-							<h2 className="glow-text-white">{name}</h2>
-							<span>
-								{domainName.length > 0 ? `0://wilder.${domainName}` : ''}
-							</span>
-
-							<div className={styles.Price}>
-								<div>Current Price</div>
-								<span className="glow-text-white">W1.56 </span>
-								<span className="glow-text-white">($8000)</span>
-							</div>
-							<Member
-								id={creatorId}
-								name={randomName(creatorId)}
-								image={randomImage(creatorId)}
-								subtext={'Creator'}
-							/>
-						</div>
-					</div>
-
-					<div className={styles.InputWrapper}>
-						<p className="glow-text-blue">
-							Enter the wallet address to transfer to
-						</p>
-						<div style={{ display: 'flex', width: '100%' }}>
-							<div className={styles.Inputs}>
-								<EtherInput
-									ethlogo
-									text={walletAddress}
-									onChange={(text: string) => setWalletAddress(text)}
-									placeholder="Ethereum Wallet"
-									type="text"
-								/>
-							</div>
-						</div>
-					</div>
+	const steps = {
+		[Step.Details]: (
+			<NFTDetails
+				creatorId={creatorId}
+				domainName={domainName}
+				title={name}
+				image={image}
+				valid={valid}
+				walletAddress={walletAddress}
+				setWalletAddress={setWalletAddress}
+				onNext={() => valid && onAccept()}
+			/>
+		),
+		[Step.Confirmation]: isLoading ? (
+			<Wizard.Loading
+				message={constants.MESSAGES.TEXT_CONFIRMATION}
+				subtext={constants.MESSAGES.TEXT_ACCEPT_PROMPT}
+			/>
+		) : (
+			<Wizard.Confirmation
+				error={error}
+				message={constants.MESSAGES.TEXT_CONFIRMATION}
+				primaryButtonText={constants.BUTTONS[Step.Confirmation].PRIMARY}
+				onClickPrimaryButton={submitTransfer}
+				secondaryButtonText={constants.BUTTONS[Step.Confirmation].SECONDARY}
+				onClickSecondaryButton={onClose}
+			/>
+		),
+	};
```

Comment:
Looks good, but I prefer to use `useMemo` to memoize

Reply:
I followed Bretts cancelBid Wizard for this. No preference myself, maybe we could discuss